### PR TITLE
Primary Selinux policy for tendrl

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Initial selinux policy for tendrl server to run in permissive domain
This provides the following features:
   - Allows tendrl processes to run in permissive mode
   - Perform checks but will not enforce
   - Users don't have to switch to permissive mode globally
   - Allows to catch AVC messages

tendrl-bug-id: Tendrl/api#279
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
This will provide selinux-carbon and tendrl-server-selinux packages

@r0h4n , @nthomas-redhat , @shtripat , Please review

**Comment by TimothyAsirJeyasing** [comment]
@Lukas Vrabec <lvrabec@redhat.com> Please review

File: selinux/tendrl.te [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+type tendrl_var_lib_t;
+files_type(tendrl_var_lib_t)
+
+type tendrl_var_run_t;
+files_pid_file(tendrl_var_run_t)
+
+type tendrl_unit_file_t;
+systemd_unit_file(tendrl_unit_file_t)
+
+type collectd_conf_t;
```
Comment (wrabcak) [inline selinux/tendrl.te]: This type definition should be in collectd SELinux module.

File: selinux/tendrl.te [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	sysnet_exec_ifconfig(collectd_t)
+	corecmd_exec_bin(collectd_t)
+	corecmd_exec_shell(collectd_t)
+	allow tendrl_t collectd_conf_t:dir manage_dir_perms;
+	allow tendrl_t collectd_conf_t:file manage_file_perms;
+	allow collectd_t collectd_conf_t:file manage_file_perms;
+	allow collectd_t collectd_conf_t:dir manage_dir_perms;
+	allow collectd_t self:capability { setuid chown sys_resource audit_write };
+	allow collectd_t self:unix_stream_socket { connectto };
+	allow dmidecode_t collectd_t:fifo_file { getattr open read };
```
Comment (wrabcak) [inline selinux/tendrl.te]: This twho rules should be fixed in dmidecote SELinux security module

File: selinux/tendrl.te [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+corenet_tcp_bind_generic_node(tendrl_t)
+corenet_tcp_bind_http_cache_port(tendrl_t)
+corenet_tcp_connect_amanda_port(tendrl_t)
+
+optional_policy(`
+	collectd_systemctl(tendrl_t)
+')
+
+optional_policy(`
+	tendrl_stub_collectd()
```
Comment (wrabcak) [inline selinux/tendrl.te]: We don't need to stub collectd module here, you cna use collectd intrafaces and in rules where collectd_t is source context, this should be in collectd SELinux module

File: tendrl-api.spec [hunk 4]
```diff
@@ -78,10 +111,70 @@ install -Dm 0644 README.adoc Rakefile $RPM_BUILD_ROOT%{_datadir}/doc/tendrl
...
+	%{buildroot}%{_datadir}/selinux/packages
+
+# carbon
+install -p -m 644 selinux/carbon.if \
+        %{buildroot}%{_datadir}/selinux/devel/include/%{moduletype}
+install -m 0644 selinux/carbon.pp.bz2 \
+        %{buildroot}%{_datadir}/selinux/packages
+
+%post -n tendrl-server-selinux
+%_format MODULE %{_datadir}/selinux/packages/tendrl.pp.bz2
```
Comment (wrabcak) [inline tendrl-api.spec]: We have SELinux policy macros for this changes in post install phase. For more info see: https://fedoraproject.org/wiki/SELinux/IndependentPolicy#The_.25post_Section

File: selinux/tendrl.te (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply selinux/tendrl.te]: We do not have a separate policy for collectd.
Do you want to keep collectd in a separate policy?
Please suggest

File: tendrl-api.spec (hunk 4 above)
Comment (TimothyAsirJeyasing) [reply tendrl-api.spec]: Sure, Thank you, i will use those macros

File: selinux/tendrl.te (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply selinux/tendrl.te]: Sure, i will remove it from here!

File: selinux/carbon.fc [hunk 5]
```diff
@@ -11,5 +11,3 @@
 /var/log/carbon(/.*)?	gen_context(system_u:object_r:carbon_log_t,s0)
 
 /var/run/carbon-(aggregator|cache)\.pid		gen_context(system_u:object_r:carbon_var_run_t,s0)
-
-/var/log/graphite-web(/.*)?     gen_context(system_u:object_r:httpd_log_t,s0)
```
Comment (wrabcak) [inline selinux/carbon.fc]: griphite_web should have graphite_log_t SELinux context

**Comment by wrabcak** [comment]
SELinux review done.

**Comment by nthomas-redhat** [comment]
Please link with git issue so that we can merge this

**Pull request merged**